crates/pyhq/src/client/job.rs

```diff
@@ -92,7 +93,15 @@ pub fn submit_job_impl(py: Python, ctx: ClientContextPtr, job: PyJobDescription)
             rpc_call!(ctx.session.connection(), message, ToClientMessage::SubmitResponse(r) => r)
                 .await
                 .map_py_err()?;
-        Ok(response.job.info.id.as_num())
+        match response {
+            SubmitResponse::Ok { job, .. } => Ok(job.info.id.as_num()),
+            SubmitResponse::JobNotOpened
+            | SubmitResponse::JobNotFound
+            | SubmitResponse::TaskIdAlreadyExists(_) => {
+                // This code is unreachable as long Python cannot submit into open jobs
+                unreachable!()
```

Comment:
Looking forward to hitting this `unreachable` block in the future :laughing:

Reply:
You are welcome :)